fix(tui): Remove unused React import in ChatMessage.tsx

## Description
PR #1008 (TUI memoization) introduced an unused React import that breaks TypeScript compilation.

## Error
```
src/components/ChatMessage.tsx(1,8): error TS6133: 'React' is declared but its value is never read.
```

## Fix
Remove unused `React` import from ChatMessage.tsx line 1.

## Priority
P0 - Blocks TUI build

--- a/tui/src/components/ChatMessage.tsx
+++ b/tui/src/components/ChatMessage.tsx
@@ -1,4 +1,4 @@
-import React, { memo } from 'react';
+import { memo } from 'react';
 import { Box, Text } from 'ink';
 import { MentionText } from './MentionText';
 import { ReactionBar } from './Reaction';

